src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java

```diff
@@ -108,13 +109,42 @@ public Custody updateCustody(final String nomsNumber,
         });
     }
 
-
+    @Transactional
     public Custody updateCustodyBookingNumber(String nomsNumber, UpdateCustodyBookingNumber updateCustodyBookingNumber) {
-        log.info("Request to update booking number for {}", nomsNumber);
-        return Custody
-                .builder()
-                .bookingNumber(updateCustodyBookingNumber.getBookingNumber())
-                .build();
+        final var maybeOffender = offenderRepository.findByNomsNumber(nomsNumber);
+        final var telemetryProperties = Map.of("offenderNo", nomsNumber,
+                "bookingNumber", updateCustodyBookingNumber.getBookingNumber(),
+                "sentenceStartDate", updateCustodyBookingNumber.getSentenceStartDate().format(DateTimeFormatter.ISO_DATE));
+
+        return maybeOffender.map(offender -> {
+            try {
+                final var maybeEvent = convictionService.getSingleActiveConvictionIdByOffenderIdAndCloseToSentenceDate(offender.getOffenderId(), updateCustodyBookingNumber.getSentenceStartDate());
+                return maybeEvent.map(event -> {
+                    // update booking Number
```

Comment:
All the TOD bits in comments, so service still does nothing

Reply:
Perhaps include TODO in the comment?  Just thinking if you want to search for outstanding stuff.  IDE's can then help out hear searching or even grepping for example....just a thought.

Reply:
Good, point, although I'm doing the TODOs as we speak